kolibri/plugins/learn/assets/src/composables/__tests__/useCopies.spec.js

```diff
@@ -0,0 +1,22 @@
+import useCopies from '../useCopies';
+
+describe('useCopies', () => {
+  it('should default to an empty array', () => {
+    const { displayedCopies } = useCopies();
+    expect(displayedCopies.copies).toEqual([]);
+  });
+  it('should set displayed copies to value passed to setCopies function', () => {
+    const { displayedCopies, setCopies } = useCopies();
+    setCopies([{ id: '1' }, { id: '2' }]);
+    expect(displayedCopies.copies).toEqual([{ id: '1' }, { id: '2' }]);
+  });
+  it('displayedCopies should be a module-level store, so the previous test should still pass', () => {
```

Comment:
I'm not entirely sure off hand - but I believe that once you import something like `useCopies` into your file, its module-level scope is persisted for the duration of the running of that file, in this case the entire test suite.

So, since `useCopies` is a function that returns a module-level `reactive()` variable like `displayedCopies`, I expect that that value should persist across all tests, so any time you need it to be a value, you need to set it anew.

If that is correct, then if I were testing a component using `useCopies` and wanted to mock it, then I'd want to `jest.resetAllMocks` in a `beforeEach` if I needed each test to presume the original state or if I wanted to ensure a specific state for each pass like:

```
jest.mock('./useFns');
// ... //
// later on in a describe() block
var { setFn, reactiveValue } = useFns();

beforeEach(() => {
    jest.resetAllMocks();
    setFn({ value: 'that I want to initialize in each test in this describe block' });
}

// then my tests that depend on that initial value in this test case
```

(cc @rtibbles does this all sound right?)

Reply:
This is all great feedback, thank you all.

I did it how I did to test the fact that the state was indeed outside of the function scope, but I could have tested that within the same `it` block.

I think that with these composables, using the mocks and the `resetAllMocks` function makes it clear when/how the state is being reset - but in the case above this isn't immediately clear.

In any case, I'll be refactoring this out but will definitely take this with me as I continue to develop test suites.